packages/vite/src/node/plugins/resolve.ts

```diff
@@ -506,8 +512,10 @@ function tryResolveFile(
           }
         }
       }
-      const index = tryFsResolve(file + '/index', options)
-      if (index) return index + postfix
+      const indexFile = tryIndexFile(file, targetWeb, options)
```

Comment:
This change avoids checking for `/index/index` or `index/package.json`, both of which would be incorrect if found.